pyccel/codegen/printing/cwrappercode.py

```diff
@@ -330,6 +333,202 @@ def optional_element_management(self, used_names, a, collect_var):
         (LiteralTrue(), [Assign(VariableAddress(a), a.value)]))]
         return optional_tmp_var, body
 
+    def _print_Interface(self, expr):
+
+        #Collecting all functions names
+        #TO DEBUG remove this line and uncomment the next one
+        funcs = [expr]
+        #funcs = expr.functions
+        # Save all used names
+        used_names = set(n.name for n in funcs)
+
+        # Find a name for the wrapper function
+        # TO DEBUG remove this line  later
+        wrapper_name = self.get_new_name(used_names.union(self._global_names), expr.name.name+"_wrapper")
+        #wrapper_name = self.get_new_name(used_names.union(self._global_names), expr.name+"_wrapper")
+        self._function_wrapper_names[expr.name] = wrapper_name
+        self._global_names.add(wrapper_name)
+
+        # Collect local variables
+        python_func_args    = self.get_new_PyObject("args"  , used_names)
+        python_func_kwargs  = self.get_new_PyObject("kwargs", used_names)
+        python_func_selfarg = self.get_new_PyObject("self"  , used_names)
+
+        # Collect wrapper arguments and results
+        wrapper_args    = [python_func_selfarg, python_func_args, python_func_kwargs]
+        wrapper_results = [self.get_new_PyObject("result", used_names)]
+
+        # Collect parser arguments
+        parse_args = [Variable(dtype=PyccelPyObject(), is_pointer=True, name = self.get_new_name(used_names, a.name+"_tmp")) for a in funcs[0].arguments]
+        wrapper_vars = {a.name : a for a in parse_args}
+
+        # Collect argument names for PyArgParse
+        arg_names         = [a.name for a in funcs[0].arguments]
+        keyword_list_name = self.get_new_name(used_names,'kwlist')
+        keyword_list      = PyArgKeywords(keyword_list_name, arg_names)
+        wrapper_body      = [keyword_list]
+
+        wrapper_body_translations = []
+        body_tmp = []
+
+        # To store the mini function responsible of collecting value and calling interfaces functions and return the builded value
+        funcs_def = []
+        default_value = {} # dict to collect all initialisation needed in the wrapper
+        check_var = Variable(dtype = NativeInteger(), name = self.get_new_name(used_names , "check"))
+        wrapper_vars[check_var.name] = check_var
+        types_dict = {} #dict to collect each variable possible type and the corresponding flags
+        # Managing the body of wrapper
+        # TODO split or re use exisiting functions in the wrapper
+        for func in funcs :
+            mini_wrapper_func_body = []
+            res_args = []
+            mini_wrapper_func_vars = [a for a in func.arguments]
```

Comment:
Use
`func.arguments.copy()`
or
`list(func.arguments)`

Reply:
i think  mini_wrapper_func_vars = func.arguments is better .

Reply:
As long as you don't modify the list then yes